jumanji/training/networks/robot_warehouse/actor_critic.py

```diff
@@ -0,0 +1,169 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Optional, Sequence
+
+import chex
+import haiku as hk
+import jax.numpy as jnp
+import numpy as np
+
+from jumanji.environments import RobotWarehouse
+from jumanji.environments.routing.robot_warehouse.types import Observation
+from jumanji.training.networks.actor_critic import (
+    ActorCriticNetworks,
+    FeedForwardNetwork,
+)
+from jumanji.training.networks.parametric_distribution import (
+    MultiCategoricalParametricDistribution,
+)
+from jumanji.training.networks.transformer_block import TransformerBlock
+
+
+def make_actor_critic_networks_robot_warehouse(
+    robot_warehouse: RobotWarehouse,
+    transformer_num_blocks: int,
+    transformer_num_heads: int,
+    transformer_key_size: int,
+    transformer_mlp_units: Sequence[int],
+) -> ActorCriticNetworks:
+    """Make actor-critic networks for the `RobotWarehouse` environment."""
+    num_values = np.asarray(robot_warehouse.action_spec().num_values)
+    parametric_action_distribution = MultiCategoricalParametricDistribution(
+        num_values=num_values
+    )
+    policy_network = make_actor_network(
+        time_limit=robot_warehouse.time_limit,
+        transformer_num_blocks=transformer_num_blocks,
+        transformer_num_heads=transformer_num_heads,
+        transformer_key_size=transformer_key_size,
+        transformer_mlp_units=transformer_mlp_units,
+    )
+    value_network = make_critic_network(
+        time_limit=robot_warehouse.time_limit,
+        transformer_num_blocks=transformer_num_blocks,
+        transformer_num_heads=transformer_num_heads,
+        transformer_key_size=transformer_key_size,
+        transformer_mlp_units=transformer_mlp_units,
+    )
+    return ActorCriticNetworks(
+        policy_network=policy_network,
+        value_network=value_network,
+        parametric_action_distribution=parametric_action_distribution,
+    )
+
+
+class RobotWarehouseTorso(hk.Module):
+    def __init__(
+        self,
+        transformer_num_blocks: int,
+        transformer_num_heads: int,
+        transformer_key_size: int,
+        transformer_mlp_units: Sequence[int],
+        env_time_limit: int,
+        name: Optional[str] = None,
+    ):
+        super().__init__(name=name)
+        self.transformer_num_blocks = transformer_num_blocks
+        self.transformer_num_heads = transformer_num_heads
+        self.transformer_key_size = transformer_key_size
+        self.transformer_mlp_units = transformer_mlp_units
+        self.model_size = transformer_num_heads * transformer_key_size
+        self.embedder = hk.Linear(self.model_size)
```

Comment:
Optional: since this is used only once, it could directly be defined where it is used (see below).
```suggestion
```